### PR TITLE
Add RankNetLoss

### PR DESCRIPTION
**Pull request by yjoonjang**
# RankNetLoss Implementation for Cross Encoder Trainer
This PR adds RankNetLoss functionality to the Cross Encoder Trainer feature.

## Changes
- Implemented RankNetLoss as a pairwise loss function
- Added RankNetLoss to class LambdaLoss with NoWeightingScheme
- Created an example script demonstrating RankNetLoss usage
## Implementation Details
RankNetLoss is a pairwise loss function which learns a ranking function by optimizing pairwise document comparisons using a neural network.
The implementation inherits from the LambdaLoss base class and uses a NoWeightingScheme, focusing solely on the pairwise comparison of documents. 

The loss function implements the core RankNet algorithm from the Burges et al. paper (2005), which uses gradient descent to learn a ranking function by optimizing a probabilistic cost function based on pairwise document comparisons.

Reference: https://icml.cc/Conferences/2015/wp-content/uploads/2015/06/icml_ranking.pdf

### CODE REVIEW

**Comment by yjoonjang** [comment]
https://github.com/UKPLab/sentence-transformers/pull/3222#issuecomment-2737453230

**Comment by yjoonjang** [comment]
Hi @tomaarsen, I've added RankNetLoss. 
Would be grateful if you could take a look at it!

**Comment by tomaarsen** [comment]
I think this is looking great! I'm merging it now 🤗 
Thanks a bunch. Hopefully I can push out a release with all of this soon!

- Tom Aarsen

**Comment by yjoonjang** [comment]
Thank you and great work!!
